snapcraft/internal/deltas/_deltas.py

```diff
@@ -0,0 +1,211 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import logging
+import os
+import subprocess
+import time
+
+from progressbar import ProgressBar
+
+from snapcraft import file_utils
+from snapcraft.internal.deltas.errors import (
+    DeltaFormatError,
+    DeltaFormatOptionError,
+    DeltaGenerationError,
+    DeltaToolError,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+delta_formats_options = [
+    'xdelta'
+]
+
+
+class BaseDeltasGenerator:
+    """Class for delta generation
+
+    This class is responsible for the snap delta file generation
+    """
+
+    def __init__(self, *, source_path, target_path,
+                 delta_file_extname='delta', delta_format=None,
+                 delta_tool_path=None):
+
+        self.source_path = source_path
+        self.target_path = target_path
+        self.delta_format = delta_format
+        self.delta_file_extname = delta_file_extname
+        self.delta_tool_path = delta_tool_path
+
+        # some pre-checks
+        self._check_properties()
+        self._check_file_existence()
+        self._check_delta_gen_tool()
+
+    def _check_properties(self):
+        if not self.delta_format:
+            raise DeltaFormatError()
+        if not self.delta_tool_path:
+            raise DeltaToolError()
+        if self.delta_format not in delta_formats_options:
+            raise DeltaFormatOptionError(
+                delta_format=self.delta_format,
+                format_options_list=delta_formats_options)
+
+    def _check_file_existence(self):
+        if not os.path.exists(self.source_path):
+            raise ValueError(
+                'source file {!r} not exist'.format(self.source_path))
+
+        if not os.path.exists(self.target_path):
+            raise ValueError(
+                'target file {!r} not exist'.format(self.target_path))
```

Comment:
updated to: 
The target file {!r} does not exist, please specify a valid target file.